### PR TITLE
Add docs to set permissions due to Kubernetes bug

### LINKED ISSUE #1157: service-binding-operator - crunchyDB operator populating tls.key with wrong permissions
This issue is not related 100% to service-binding-operator

### What is the environment (Minikube, Openshift)?
Openshift

### What is the SBO version used?
apiVersion: binding.operators.coreos.com/v1alpha1

### What are the steps to reproduce this issue?
  1.  Deploying [cruncry db](https://access.crunchydata.com/documentation/postgres-operator/5.1.0/tutorial/) in openshift
  2. Deploying a simple nodejs app
  3. Using a service binding operator to bind the app with node js app
  4. Both services are being bound successfully after creating service binding, and on env variables of nodejs app, a SERVICE_BINDING_ROOT env has been set.
  5. under SERVICE_BINDING_ROOT/myapp are all files populated from crunchy db and ready to use, in order to connect with the Database.

### What is the expected behaviour?
Under SERVICE_BINDING_ROOT/myapp directory, having tls.key with proper permissions which in our case is 0600

### What is the actual behaviour?
tls.key has user or world group permissions.

### Postresql logs
Postgresql throws below error:
```
private key file "/bindings/myapp/tls.key" has group or world access; permissions should be u=rw (0600) or less\n
```

### PR DESCRIPTION
**Pull request by baijum**
Fixes #1157

Signed-off-by: Baiju Muthukadan <baiju.m.mail@gmail.com>

# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] [Docs](https://github.com/redhat-developer/service-binding-operator/blob/master/CONTRIBUTING.md#docs) 
  included if any changes are user facing
- [ ] [Tests](https://github.com/redhat-developer/service-binding-operator/blob/master/CONTRIBUTING.md#tests)
  included if any functionality added or changed. For bugfixes please include tests that can catch regressions
- [ ] All acceptance test scenarios included in the PR which verifies a bugfix or a requested feature reported by a non-member are tagged with `@external-feedback` tag.
- [ ] Follows the [commit message standard](https://github.com/redhat-developer/service-binding-operator/blob/master/CONTRIBUTING.md#commits)

### CODE REVIEW

**Review by sadlerap — COMMENTED** [review]
With this fix, lgtm.
Would like a review from @Srivaralakshmi before merging.

File: docs/userguide/modules/projecting-binding-data/pages/projecting-binding-data.adoc [hunk 1]
```diff
@@ -45,7 +45,7 @@ Retrieving the binding data through the binding data directory name is not a goo
 
 [WARNING]
 ====
-**For using the binding data directory name to look up the binding data:** The {servicebinding-title} uses the `ServiceBinding` resource name (.`metadata.name`) as the binding data directory name, but the spec also provides a way to override that name through the `.spec.name` field. Hence there is a chance for binding data name collision. However, due to the nature of the volume mount in Kubernetes, the binding data directory will contain values from only one of the `Secret` resources. 
+**For using the binding data directory name to look up the binding data:** The {servicebinding-title} uses the `ServiceBinding` resource name (.`metadata.name`) as the binding data directory name, but the spec also provides a way to override that name through the `.spec.name` field. Hence there is a chance for binding data name collision. However, due to the nature of the volume mount in Kubernetes, the binding data directory will contain values from only one of the `Secret` resources.
```
[the remark this answers is not in the document]
Comment (baijum) [reply docs/userguide/modules/projecting-binding-data/pages/projecting-binding-data.adoc:48]: @sadlerap 
This change was an accidental change. My editor removed the extra whitespace at the end of the lines.

I hope you must be checked the other note.

File: docs/userguide/modules/projecting-binding-data/pages/projecting-binding-data.adoc [hunk 2]
```diff
@@ -137,3 +137,8 @@ Depending on your workload requirements and environment, you can choose to proje
 
 . To project the binding data as files, determine the destination folder by ensuring that the existing `SERVICE_BINDING_ROOT` environment variable is present in the container where your workload runs.
 . To project the binding data as environment variables, set the value for the `.spec.bindAsFiles` parameter to `false` from the `ServiceBinding` resource in the custom resource (CR).
+
+[NOTE]
+====
+The projected files gets the default permission.  If the service expects a specific permission, say `0600`, that's not possible due [a bug in Kubernetes](https://github.com/kubernetes/kubernetes/issues/57923).  As a workaround, the application can copy the file to `/tmp` directory and set the appropriate permissions.
```
Comment (sadlerap) [inline docs/userguide/modules/projecting-binding-data/pages/projecting-binding-data.adoc:143]: I think I like this wording a bit better.
```suggestion
By default, the projected files get their permissions set to 0644.  {servicebinding-title} cannot set specific permissions due to [a bug in Kubernetes](https://github.com/kubernetes/kubernetes/issues/57923), which causes issues if the service expects a specific permission, such as `0600`.  As a workaround, the application can copy the file to `/tmp` directory and set the appropriate permissions.
```

**Review by sadlerap — COMMENTED** [review]
Would like a review from @Srivaralakshmi as well.

File: docs/userguide/modules/projecting-binding-data/pages/projecting-binding-data.adoc [hunk 3]
```diff
@@ -137,3 +137,8 @@ Depending on your workload requirements and environment, you can choose to proje
 
 . To project the binding data as files, determine the destination folder by ensuring that the existing `SERVICE_BINDING_ROOT` environment variable is present in the container where your workload runs.
 . To project the binding data as environment variables, set the value for the `.spec.bindAsFiles` parameter to `false` from the `ServiceBinding` resource in the custom resource (CR).
+
+[NOTE]
+====
+By default, the projected files get their permissions set to 0644.  {servicebinding-title} cannot set specific permissions due to [a bug in Kubernetes](https://github.com/kubernetes/kubernetes/issues/57923), which causes issues if the service expects a specific permission, such as `0600`.  As a workaround, the application can copy the file to `/tmp` directory and set the appropriate permissions.
```
Comment (Srivaralakshmi) [inline docs/userguide/modules/projecting-binding-data/pages/projecting-binding-data.adoc:143]:
```suggestion
By default, the projected files get their permissions set to 0644.  {servicebinding-title} cannot set specific permissions due to [a bug in Kubernetes](https://github.com/kubernetes/kubernetes/issues/57923) that causes issues if the service expects specific permissions such as, `0600`.  As a workaround, you can modify the code of the program or the application that is running inside a workload resource to copy the file to the `/tmp` directory and set the appropriate permissions.
```

**Review by Srivaralakshmi — CHANGES_REQUESTED** [review]
@baijum PTAL at the suggested changes. Thanks!

**Pull request merged**